students/NatalieRodriguez/Lesson09/OO_Mailroom.py

```diff
@@ -0,0 +1,172 @@
+If a function contains a call to input() – it belongs outside of the logic classes – either stand alone in
+the module (like they are already) or perhaps all in a CLI class.'''
+
+#!/usr/bin/env python3
+import sys
+from collections import defaultdict
+
+class Donor:
+
+    def __init__(self, name, donations=None):
+
+        self.name = name
+        if donations is None:
+            self.donations = []
+        else:
+            self.donations = donations
+
+    def add_donation(self, amount):
+        try:
+            self.donations.append(int(amount))
+        except ValueError:
+            print("Enter a donation amount.")
+
+    def donation_amount(self):
+        return len(self.donations)
+
+    def total_donation(self):
+        try:
+            return sum(self.donations)
+        except TypeError:
+            return self.donations
+
+    def donation_avg(self):
+        try:
+            return self.total_donation()/ self.donation_amount()
+        except TypeError:
+            return self.donations
+
+class DonorHistory:
+
+    def __init__(self, donors=None):
+        self.donor_list = donors
+
+    def thank_you(self):
+        '''user inputs donor and amounts to dict.
+        '''
+        print('Type the full donor name (Enter "list" to show all donors or enter "dash" to return to the Donor Dashboard.')
+        new_donor = input(':')
+
+        if new_donor.lower() == 'list':
+            donor_names = []
+            for donor in self.donor_list:
+                donor_names.append(donor.name)
+            print(donor_names)
+
+        elif new_donor.lower() == 'dash':
+            return
+
+        else:
+            if new_donor in self.donor_list:
+                try:
+                    amount = float(input('Please enter the donation amount:'))
+                    donor_list[Donor].add_donation(amount)
+
+                except ValueError:
+                    print("Please enter a valid amount.")
+            else:
+                amount = float(input('Please enter the donation amount:'))
+                new_donor_object = Donor(new_donor, [amount])
+                self.donor_list.append(new_donor_object)
+                print('            Thank you {}, We greatly appreciate your generous donation of ${:.2f}.'.format(new_donor, amount))
+                print('''            Your contribution to the purchase of land for conservation
+            and habitat restoration will make a profound difference in allowing
+            the environment to recover from the detrimental impact of human
+            life. With noble individuals like yourself, we can make large strides
+            to allow the North American continent to exist without the stressors of 
+            agriculture and other human impacts. We appreciate and look forward to
+            your continued support.
+            
+            Best,
+            The Nature Conservancy''')
```

Comment:
this letter text would be better in Donor class since it only deals with single donor, ie:
```
new_donor_object.get_letter()
```